tests/system/src/test/java/test/javafx/scene/control/ProgressIndicatorLeakTest.java

```diff
@@ -0,0 +1,131 @@
+ */
+
+package test.javafx.scene.control;
+
+import javafx.application.Application;
+import javafx.application.Platform;
+import javafx.scene.Group;
+import javafx.scene.Node;
+import javafx.scene.Parent;
+import javafx.scene.Scene;
+import javafx.scene.control.ProgressIndicator;
+import javafx.scene.control.skin.ProgressIndicatorSkin;
+import javafx.stage.Stage;
+import javafx.stage.WindowEvent;
+
+import java.lang.ref.WeakReference;
+import java.lang.reflect.Field;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import junit.framework.Assert;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class ProgressIndicatorLeakTest {
+
+    static CountDownLatch startupLatch;
+    static WeakReference<Node> detIndicator = null;
+    static Stage stage = null;
+
+    public static class TestApp extends Application {
+
+        @Override
+        public void start(Stage primaryStage) throws Exception {
+            ProgressIndicator indicator = new ProgressIndicator(-1);
+            indicator.setSkin(new ProgressIndicatorSkin(indicator));
+            Scene scene = new Scene(indicator);
+            primaryStage.setScene(scene);
+            stage = primaryStage;
+            indicator.setProgress(1.0);
+            Assert.assertEquals("size is wrong", 1, indicator.getChildrenUnmodifiable().size());
+            detIndicator = new WeakReference<Node>(indicator.getChildrenUnmodifiable().get(0));
+            indicator.setProgress(-1.0);
+            indicator.setProgress(1.0);
+
+            stage.addEventHandler(WindowEvent.WINDOW_SHOWN, e -> {
+                Platform.runLater(() -> {
+                    startupLatch.countDown();
+                });
+            });
+            primaryStage.show();
+        }
+    }
+
+    @BeforeClass
+    public static void initFX() throws Exception {
+        startupLatch = new CountDownLatch(1);
+        new Thread(() -> Application.launch(TestApp.class, (String[]) null)).start();
+
+        Assert.assertTrue("Timeout waiting for FX runtime to start", startupLatch.await(15, TimeUnit.SECONDS));
+    }
+
+
+    @Test
+    public void memoryTest() throws Exception {
+        assertCollectable(detIndicator);
+    }
+
+    public static void assertCollectable(WeakReference weakReference) throws Exception {
+        int counter = 0;
+
+        createGarbage();
+        System.gc();
+        System.runFinalization();
+
+        while (counter < 10 && weakReference.get() != null) {
+            Thread.sleep(100);
+            counter = counter + 1;
+            createGarbage();
+            System.gc();
+            System.runFinalization();
+        }
+
+        if (weakReference.get() != null) {
+            throw new AssertionError("Content of WeakReference was not collected. content: " + weakReference.get());
+        }
+    }
+    public static void createGarbage() {
```

Comment:
I think the test should consistently pass or fail without using this method.
Can you please check and remove this method if can be.
I tested once and it passes.

Reply:
It's quite dependent on how the JVM and the GC are implemented.
It might be only really required when elements get into a very old generation of the heap.
I remember that I required this method to make similar tests stable.
Nevertheless, I've now removed the method. This way the test is also more similar to the other memory tests in the jfx project..